### PR TITLE
SDK-1470: Move Sandbox

### PR DESCRIPTION
**Pull request by davidgrayston**
### Added
- Sandbox Service
  - `Yoti::Sandbox::Profile::Client#setup_sharing_profile` to interact with the Sandbox service and create a token for the specified profile

#### Profile Token Creation

```ruby
require 'yoti'
require 'yoti_sandbox'
require 'base64'

Yoti.configure do |config|
  config.client_sdk_id = 'SANDBOX_CLIENT_SDK_ID'
  config.key_file_path = '/path/to/your-pem-file.pem'
  config.api_endpoint = 'https://api.yoti.com/sandbox/v1'
end

age_verification = Yoti::Sandbox::Profile::AgeVerificationBuilder
  .new
  .with_date_of_birth(DateTime.new(1989, 1, 2))
  .with_age_over(18)
  .build

token_request = Yoti::Sandbox::Profile::TokenRequestBuilder
  .new
  .with_remember_me_id('some_remember_me_id')
  .with_given_names('Some Given Names')
  .with_family_name('Some Family Name')
  .with_full_name('Some Full Name')
  .with_date_of_birth(DateTime.new(1989, 1, 2))
  .with_age_verification(age_verification)
  .with_gender('Some Gender')
  .with_phone_number('Some Phone Number')
  .with_nationality('Some Nationality')
  .with_postal_address('Some Postal Address')
  .with_structured_postal_address({
    building_number: 1,
    address_line1: 'Some Address'
  })
  .with_base64_selfie(Base64.strict_encode64('Some Base64 Encoded Selfie'))
  .with_email_address('Some Email Address')
  .with_document_details('PASSPORT USA 1234abc')
  .build

sandboxClient = Yoti::Sandbox::Profile::Client.new
token = sandboxClient.setup_sharing_profile(token_request)

activityDetails = Yoti::Client.get_activity_details(token)
```

### Changed
- Moved all classes into `Yoti::Sandbox::Profile` module
- Default `Yoti::Sandbox::Profile::Client` _base_url_ is now https://api.yoti.com/sandbox/v1 - _base_url_ is now optional
- Renamed `Sandbox::Profile` to `Yoti::Sandbox::Profile::TokenRequestBuilder`, which builds `TokenRequest`
- `Yoti::Sandbox::Profile::Anchor` now accepts timestamp as `DateTime` or `Time`

### Added
- `Yoti::Sandbox::Profile::AgeVerification`
- `Yoti::Sandbox::Profile::AgeVerificationBuilder`
- `Yoti::Sandbox::Profile::TokenRequest`
- `Yoti::Sandbox::Profile::TokenRequestBuilder#with_base64_selfie`

### Fixed
- Date of Birth now formatted as `YYYY-MM-DD` (was including time)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,2 +1,77 @@
...
+
+Or simply run the following command from your terminal:
+
+```shell
+gem install yoti_sandbox
+```
+
+## Configuration
+
+* `CLIENT_SDK_ID` is the SDK identifier generated by Yoti Hub in the Key tab when you create your app.
```
Comment (echarrod) [inline README.md]: Would be good to distinguish between regular and Sandbox SDK IDs. I realise it's like this in other ones too, so if this sounds ok we can update the rest
```suggestion
* `CLIENT_SDK_ID` is the Sandbox SDK identifier generated by Yoti Hub from the Sandbox section. This is different to the Client SDK ID for your application.
```

Comment (echarrod) [reply README.md]: Could even update the var name to `SANDBOX_CLIENT_SDK_ID`? As I know people won't always read the READMEs. @jlmartyres what do you think?

**Review by echarrod — CHANGES_REQUESTED** [review]

File: README.md (hunk 1 above)
Comment (jlmartyres) [reply README.md]: That's a good idea, and SANDBOX_CLIENT_SDK_ID is consistent with the Hub

Comment (davidgrayston) [reply README.md]: Updated in 5336112

**Review by echarrod — APPROVED** [review]
👌

**Comment by davidgrayston** [comment]
@echarrod optional has been removed in 03c2685

**Review by echarrod — APPROVED** [review]

**Review by MrBurtyyy — APPROVED** [review]

**Pull request merged**